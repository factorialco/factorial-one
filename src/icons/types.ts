import { ForwardRefExoticComponent, RefAttributes } from "react";
import { SvgProps } from "react-native-svg";

export type IconComponent = ForwardRefExoticComponent<
  SvgProps &
    RefAttributes<SVGSVGElement> & {
      className?: string;
    }
>;
