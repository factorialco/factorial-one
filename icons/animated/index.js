export { default as BarGraphAnimated } from "./BarGraph";
export { default as BookAnimated } from "./Book";
export { default as BuildingAnimated } from "./Building";
export { default as CalendarAnimated } from "./Calendar";
export { default as CheckCircleAnimated } from "./CheckCircle";
export { default as CheckCircleLineAnimated } from "./CheckCircleLine";
export { default as ClockAnimated } from "./Clock";
export { default as DesktopAnimated } from "./Desktop";
export { default as FlagAnimated } from "./Flag";
export { default as FoldersAnimated } from "./Folders";
export { default as FolderUserAnimated } from "./FolderUser";
export { default as GraphAnimated } from "./Graph";
export { default as HeartAnimated } from "./Heart";
export { default as HomeAnimated } from "./Home";
export { default as InboxAnimated } from "./Inbox";
export { default as MessageHeartAnimated } from "./MessageHeart";
export { default as MoneyAnimated } from "./Money";
export { default as MoneyBagAnimated } from "./MoneyBag";
export { default as PalmTreeAnimated } from "./PalmTree";
export { default as PeopleAnimated } from "./People";
export { default as PersonAnimated } from "./Person";
export { default as RocketAnimated } from "./Rocket";
export { default as ScheduleAnimated } from "./Schedule";
export { default as SearchPersonAnimated } from "./SearchPerson";
export { default as SettingsAnimated } from "./Settings";
export { default as SparklesAnimated } from "./Sparkles";
export { default as SplitAnimated } from "./Split";
export { default as SuitcaseAnimated } from "./Suitcase";
export { default as TimerAnimated } from "./Timer";
export { default as WalletAnimated } from "./Wallet";
