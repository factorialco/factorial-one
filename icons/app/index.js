export { default as AcademicCap } from "./AcademicCap";
export { default as Add } from "./Add";
export { default as Ai } from "./Ai";
export { default as Alert } from "./Alert";
export { default as AlertCircle } from "./AlertCircle";
export { default as AlertCircleLine } from "./AlertCircleLine";
export { default as AlignTextCenter } from "./AlignTextCenter";
export { default as AlignTextJustify } from "./AlignTextJustify";
export { default as AlignTextLeft } from "./AlignTextLeft";
export { default as AlignTextRight } from "./AlignTextRight";
export { default as Appearance } from "./Appearance";
export { default as Archive } from "./Archive";
export { default as ArchiveOpen } from "./ArchiveOpen";
export { default as ArrowCycle } from "./ArrowCycle";
export { default as ArrowDown } from "./ArrowDown";
export { default as ArrowLeft } from "./ArrowLeft";
export { default as ArrowRight } from "./ArrowRight";
export { default as ArrowUp } from "./ArrowUp";
export { default as Balance } from "./Balance";
export { default as Bank } from "./Bank";
export { default as BarGraph } from "./BarGraph";
export { default as Bell } from "./Bell";
export { default as Bold } from "./Bold";
export { default as BookOpen } from "./BookOpen";
export { default as Briefcase } from "./Briefcase";
export { default as Bucket } from "./Bucket";
export { default as Building } from "./Building";
export { default as Bullet } from "./Bullet";
export { default as Calculator } from "./Calculator";
export { default as Calendar } from "./Calendar";
export { default as CalendarArrowDown } from "./CalendarArrowDown";
export { default as CalendarArrowLeft } from "./CalendarArrowLeft";
export { default as CalendarArrowRight } from "./CalendarArrowRight";
export { default as CameraPlus } from "./CameraPlus";
export { default as CardPin } from "./CardPin";
export { default as ChartLine } from "./ChartLine";
export { default as ChartPie } from "./ChartPie";
export { default as Check } from "./Check";
export { default as CheckCircle } from "./CheckCircle";
export { default as CheckCircleLine } from "./CheckCircleLine";
export { default as CheckDouble } from "./CheckDouble";
export { default as ChevronDown } from "./ChevronDown";
export { default as ChevronLeft } from "./ChevronLeft";
export { default as ChevronRight } from "./ChevronRight";
export { default as ChevronUp } from "./ChevronUp";
export { default as Circle } from "./Circle";
export { default as Clock } from "./Clock";
export { default as ClockBack } from "./ClockBack";
export { default as Cloud } from "./Cloud";
export { default as Code } from "./Code";
export { default as Coffee } from "./Coffee";
export { default as Command } from "./Command";
export { default as Comment } from "./Comment";
export { default as Completed } from "./Completed";
export { default as Contactless } from "./Contactless";
export { default as CornerHandle } from "./CornerHandle";
export { default as CreditCard } from "./CreditCard";
export { default as Cross } from "./Cross";
export { default as CrossedCircle } from "./CrossedCircle";
export { default as Crown } from "./Crown";
export { default as Delete } from "./Delete";
export { default as Deny } from "./Deny";
export { default as Desktop } from "./Desktop";
export { default as DollarBill } from "./DollarBill";
export { default as DottedCircle } from "./DottedCircle";
export { default as Download } from "./Download";
export { default as DropdownDefault } from "./DropdownDefault";
export { default as DropdownOpen } from "./DropdownOpen";
export { default as Ellipsis } from "./Ellipsis";
export { default as EllipsisHorizontal } from "./EllipsisHorizontal";
export { default as Envelope } from "./Envelope";
export { default as EnvelopeOpen } from "./EnvelopeOpen";
export { default as Equal } from "./Equal";
export { default as Exit } from "./Exit";
export { default as ExternalLink } from "./ExternalLink";
export { default as EyeInvisible } from "./EyeInvisible";
export { default as EyeVisible } from "./EyeVisible";
export { default as FaceNegative } from "./FaceNegative";
export { default as FaceNeutral } from "./FaceNeutral";
export { default as FacePositive } from "./FacePositive";
export { default as FaceSuperNegative } from "./FaceSuperNegative";
export { default as FaceSuperPositive } from "./FaceSuperPositive";
export { default as Feed } from "./Feed";
export { default as File } from "./File";
export { default as FileFilled } from "./FileFilled";
export { default as Files } from "./Files";
export { default as FileSigned } from "./FileSigned";
export { default as Filter } from "./Filter";
export { default as Flag } from "./Flag";
export { default as Folder } from "./Folder";
export { default as Folders } from "./Folders";
export { default as Globe } from "./Globe";
export { default as Graph } from "./Graph";
export { default as GroupBy } from "./GroupBy";
export { default as Handle } from "./Handle";
export { default as Handshake } from "./Handshake";
export { default as Heading1 } from "./Heading1";
export { default as Heading2 } from "./Heading2";
export { default as Heading3 } from "./Heading3";
export { default as Heart } from "./Heart";
export { default as HoldHeart } from "./HoldHeart";
export { default as Home } from "./Home";
export { default as Hub } from "./Hub";
export { default as Image } from "./Image";
export { default as Inbox } from "./Inbox";
export { default as Info } from "./Info";
export { default as InfoCircle } from "./InfoCircle";
export { default as InfoCircleLine } from "./InfoCircleLine";
export { default as InProgressTask } from "./InProgressTask";
export { default as InputField } from "./InputField";
export { default as Italic } from "./Italic";
export { default as Kanban } from "./Kanban";
export { default as Laptop } from "./Laptop";
export { default as LayersFront } from "./LayersFront";
export { default as Lightbulb } from "./Lightbulb";
export { default as Link } from "./Link";
export { default as LinkRemove } from "./LinkRemove";
export { default as List } from "./List";
export { default as LockLocked } from "./LockLocked";
export { default as LockUnlocked } from "./LockUnlocked";
export { default as LogoAvatar } from "./LogoAvatar";
export { default as LogoEruditai } from "./LogoEruditai";
export { default as LogoTravelperk } from "./LogoTravelperk";
export { default as Masonry } from "./Masonry";
export { default as Maximize } from "./Maximize";
export { default as Megaphone } from "./Megaphone";
export { default as Menu } from "./Menu";
export { default as MessageFilled } from "./MessageFilled";
export { default as MessageFrown } from "./MessageFrown";
export { default as MessageHeart } from "./MessageHeart";
export { default as Messages } from "./Messages";
export { default as Microphone } from "./Microphone";
export { default as MicrophoneNegative } from "./MicrophoneNegative";
export { default as Minimize } from "./Minimize";
export { default as Minus } from "./Minus";
export { default as Mobile } from "./Mobile";
export { default as Money } from "./Money";
export { default as MoneyBag } from "./MoneyBag";
export { default as Moon } from "./Moon";
export { default as MoveDown } from "./MoveDown";
export { default as MoveTop } from "./MoveTop";
export { default as MoveUp } from "./MoveUp";
export { default as Numbers } from "./Numbers";
export { default as Office } from "./Office";
export { default as OlList } from "./OlList";
export { default as Organization } from "./Organization";
export { default as PalmTree } from "./PalmTree";
export { default as Paperclip } from "./Paperclip";
export { default as PartiallyCompleted } from "./PartiallyCompleted";
export { default as Password } from "./Password";
export { default as PauseCircle } from "./PauseCircle";
export { default as Pencil } from "./Pencil";
export { default as People } from "./People";
export { default as Person } from "./Person";
export { default as PersonMinus } from "./PersonMinus";
export { default as PersonPlus } from "./PersonPlus";
export { default as Phone } from "./Phone";
export { default as Pig } from "./Pig";
export { default as Pin } from "./Pin";
export { default as PixBrazil } from "./PixBrazil";
export { default as Placeholder } from "./Placeholder";
export { default as Plane } from "./Plane";
export { default as Plus } from "./Plus";
export { default as Present } from "./Present";
export { default as Printer } from "./Printer";
export { default as Proyector } from "./Proyector";
export { default as Question } from "./Question";
export { default as Quote } from "./Quote";
export { default as Reaction } from "./Reaction";
export { default as Receipt } from "./Receipt";
export { default as Record } from "./Record";
export { default as RemoveFavorite } from "./RemoveFavorite";
export { default as Replace } from "./Replace";
export { default as Reset } from "./Reset";
export { default as Rocket } from "./Rocket";
export { default as Salad } from "./Salad";
export { default as Save } from "./Save";
export { default as Schedule } from "./Schedule";
export { default as Search } from "./Search";
export { default as SearchPerson } from "./SearchPerson";
export { default as Settings } from "./Settings";
export { default as Share } from "./Share";
export { default as SignPost } from "./SignPost";
export { default as Sliders } from "./Sliders";
export { default as SolidPause } from "./SolidPause";
export { default as SolidPlay } from "./SolidPlay";
export { default as SolidStop } from "./SolidStop";
export { default as Sort } from "./Sort";
export { default as Sparkles } from "./Sparkles";
export { default as Spinner } from "./Spinner";
export { default as Split } from "./Split";
export { default as Star } from "./Star";
export { default as StarFilled } from "./StarFilled";
export { default as Strikethrough } from "./Strikethrough";
export { default as Suitcase } from "./Suitcase";
export { default as Table } from "./Table";
export { default as Target } from "./Target";
export { default as TextSize } from "./TextSize";
export { default as ThumbsDown } from "./ThumbsDown";
export { default as ThumbsDownFilled } from "./ThumbsDownFilled";
export { default as ThumbsUp } from "./ThumbsUp";
export { default as ThumbsUpFilled } from "./ThumbsUpFilled";
export { default as Timer } from "./Timer";
export { default as Underline } from "./Underline";
export { default as Upload } from "./Upload";
export { default as Upsell } from "./Upsell";
export { default as Video } from "./Video";
export { default as VideoRecorder } from "./VideoRecorder";
export { default as VideoRecorderNegative } from "./VideoRecorderNegative";
export { default as Wallet } from "./Wallet";
export { default as Warning } from "./Warning";
export { default as WhatsappChat } from "./WhatsappChat";
export { default as Windows } from "./Windows";
