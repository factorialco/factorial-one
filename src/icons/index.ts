import * as AppIcons from "./app";
import * as ModuleIcons from "./modules";
export { AppIcons, ModuleIcons };
export * from "./types";
export { Icon } from "../components/Icon";
