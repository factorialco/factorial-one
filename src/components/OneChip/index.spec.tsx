import { render, screen, userEvent } from "@testing-library/react-native";
import React from "react";
import { OneChip } from ".";
import { AppIcons } from "../../icons";

describe("OneChip", () => {
  const defaultProps = {
    label: "label",
  };

  it("Snapshot", () => {
    const { toJSON } = render(<OneChip {...defaultProps} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("Snapshot variant selected", () => {
    const { toJSON } = render(<OneChip {...defaultProps} variant="selected" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("Snapshot with Icon", () => {
    const { toJSON } = render(
      <OneChip {...defaultProps} icon={AppIcons.Placeholder} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it("Snapshot with OnClose", () => {
    const handleOnClose = jest.fn();
    const { toJSON } = render(
      <OneChip {...defaultProps} onClose={handleOnClose} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders correctly with required props", () => {
    render(<OneChip {...defaultProps} />);

    const number = screen.getByText("label");

    expect(number).toBeDefined();
  });

  it("renders correctly with onClose event", async () => {
    const testFn = jest.fn();
    const user = userEvent.setup();

    render(<OneChip {...defaultProps} onClose={testFn} />);

    const close = screen.getByLabelText("Close");
    await user.press(close);

    expect(testFn).toHaveBeenCalledTimes(1);
    expect(close).toBeDefined();
  });

  it("renders correctly with onClick event", async () => {
    const testFn = jest.fn();
    const user = userEvent.setup();

    render(<OneChip {...defaultProps} onClick={testFn} />);

    const action = screen.getByLabelText("Action");
    await user.press(action);

    expect(testFn).toHaveBeenCalledTimes(1);
    expect(action).toBeDefined();
  });
});
