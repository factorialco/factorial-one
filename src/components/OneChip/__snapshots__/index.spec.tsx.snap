// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OneChip Snapshot 1`] = `
<View
  className="flex items-start"
>
  <View
    accessibilityLabel="Action"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    className="flex items-center gap-1 rounded-full border border-solid border-f1-border px-2 py-0.5 grow-0"
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      className="flex flex-row items-center gap-0.5"
    >
      <Text
        className="font-medium text-f1-foreground"
      >
        label
      </Text>
    </View>
  </View>
</View>
`;

exports[`OneChip Snapshot variant selected 1`] = `
<View
  className="flex items-start"
>
  <View
    accessibilityLabel="Action"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    className="flex items-center gap-1 rounded-full border border-solid px-2 py-0.5 grow-0 border-f1-border-selected bg-f1-background-selected-secondary"
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      className="flex flex-row items-center gap-0.5"
    >
      <Text
        className="font-medium text-f1-foreground-selected"
      >
        label
      </Text>
    </View>
  </View>
</View>
`;

exports[`OneChip Snapshot with Icon 1`] = `
<View
  className="flex items-start"
>
  <View
    accessibilityLabel="Action"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    className="flex items-center gap-1 rounded-full border border-solid border-f1-border px-2 py-0.5 grow-0 pl-1.5"
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      className="flex flex-row items-center gap-0.5"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="100%"
        bbWidth="100%"
        className="shrink-0 w-4 h-4 stroke-sm font-medium text-f1-foreground"
        fill="none"
        focusable={false}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": "100%",
              "width": "100%",
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGCircle
            cx={12}
            cy={12}
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
              ]
            }
            r={8}
            stroke={
              {
                "type": 2,
              }
            }
          />
          <RNSVGPath
            d="m17.5 6.5-11 11M17.5 17.5l-11-11"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinejoin={1}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <Text
        className="font-medium text-f1-foreground"
      >
        label
      </Text>
    </View>
  </View>
</View>
`;

exports[`OneChip Snapshot with OnClose 1`] = `
<View
  className="flex items-start"
>
  <View
    accessibilityLabel="Action"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    className="flex items-center gap-1 rounded-full border border-solid border-f1-border px-2 py-0.5 grow-0 pr-1.5"
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      className="flex flex-row items-center gap-0.5"
    >
      <Text
        className="font-medium text-f1-foreground"
      >
        label
      </Text>
      <View
        accessibilityLabel="Close"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        className="-m-1 flex h-6 w-6 cursor-pointer items-center justify-center rounded-full [&_svg]:text-f1-icon-secondary"
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        tabIndex={0}
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="100%"
          bbWidth="100%"
          className="shrink-0 w-4 h-4 stroke-sm font-medium text-f1-foreground"
          fill="none"
          focusable={false}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": "100%",
                "width": "100%",
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
              ]
            }
          >
            <RNSVGPath
              clipRule={0}
              d="M3.35 12a8.65 8.65 0 1 1 17.3 0 8.65 8.65 0 0 1-17.3 0m6.11-3.46a.65.65 0 0 0-.92.92L11.08 12l-2.54 2.54a.65.65 0 1 0 .92.92L12 12.92l2.54 2.54a.65.65 0 1 0 .92-.92L12.92 12l2.54-2.54a.65.65 0 1 0-.92-.92L12 11.08z"
              fill={
                {
                  "type": 2,
                }
              }
              fillRule={0}
              propList={
                [
                  "fill",
                  "fillRule",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
</View>
`;
